packages/backend/src/models/ModelRepository.ts

```diff
@@ -0,0 +1,465 @@
+     * ModelRepository instance.
+     *
+     * new ModelRepository({
+     *    modelProviders: {
+     *      encryptionModel: ({ repository, context }) => {
+     *          return new EncryptionModelOverride(...);
+     *      }
+     *    }
+     * })
+     *
+     * NOTE: This exact implementation is temporary, and is likely to be adjusted soon
+     * as part of the dependency injection rollout.
+     */
+    protected providers: ModelProviderMap;
+
+    protected readonly lightdashConfig: LightdashConfig;
+
+    /**
+     * Knex database instance, used for all database operations.
+     */
+    protected readonly database: Knex;
+
+    constructor({
+        modelProviders,
+        lightdashConfig,
+        database,
+    }: {
+        modelProviders?: ModelProviderMap<ModelManifest>;
+        lightdashConfig: LightdashConfig;
+        database: Knex;
+    }) {
+        this.providers = modelProviders ?? {};
+        this.lightdashConfig = lightdashConfig;
+        this.database = database;
+    }
+}
+
+/**
+ * Bare model repository class, which acts as a container for all existing
+ * models, and as a point to share instantiation and common logic.
+ *
+ * If you need to access a model, you should do it through an instance of this
+ * repository - ideally one that you accessed through a controller, or otherwise
+ * via dependency injection.
+ *
+ */
+export class ModelRepository
+    extends ModelRepositoryBase
+    implements ModelFactoryMethod<ModelManifest>
+{
+    /**
+     * Holds memoized instances of models after their initial instantiation:
+     */
+    protected modelInstances: Partial<ModelManifest> = {};
+
+    public getAnalyticsModel(): AnalyticsModel {
+        return this.getModel(
+            'analyticsModel',
+            () => new AnalyticsModel({ database: this.database }),
+        );
+    }
+
+    public getCommentModel(): CommentModel {
+        return this.getModel(
+            'commentModel',
+            () => new CommentModel({ database: this.database }),
+        );
+    }
+
+    public getDashboardModel(): DashboardModel {
+        return this.getModel(
+            'dashboardModel',
+            () => new DashboardModel({ database: this.database }),
+        );
+    }
+
+    public getDownloadFileModel(): DownloadFileModel {
+        return this.getModel(
+            'downloadFileModel',
+            () => new DownloadFileModel({ database: this.database }),
+        );
+    }
+
+    public getEmailModel(): EmailModel {
+        return this.getModel(
+            'emailModel',
+            () => new EmailModel({ database: this.database }),
+        );
+    }
+
+    public getGithubAppInstallationsModel(): GithubAppInstallationsModel {
+        return this.getModel(
+            'githubAppInstallationsModel',
+            () =>
+                new GithubAppInstallationsModel({
+                    database: this.database,
+                    encryptionService: new EncryptionService({
```

Comment:
Ideally should use the service repository to access services in this phase. We might get a circular dependency though.

Reply:
`EncryptionService` needs to be renamed `EncryptionUtils` as it is not a real service